ignite/templates/app/app.go

```diff
@@ -2,27 +2,30 @@ package app
 
 import (
 	"embed"
+	"fmt"
+	"io/fs"
 
 	"github.com/gobuffalo/genny/v2"
 	"github.com/gobuffalo/plush/v4"
 
 	"github.com/ignite/cli/ignite/pkg/cosmosgen"
 	"github.com/ignite/cli/ignite/pkg/xgenny"
 	"github.com/ignite/cli/ignite/templates/field/plushhelpers"
-	"github.com/ignite/cli/ignite/templates/testutil"
 )
 
 //go:embed files/* files/**/*
-var fs embed.FS
-
-// New returns the generator to scaffold a new Cosmos SDK app.
-func New(opts *Options) (*genny.Generator, error) {
-	var (
-		g        = genny.New()
-		template = xgenny.NewEmbedWalker(fs, "files/", opts.AppPath)
-	)
-	if err := g.Box(template); err != nil {
-		return g, err
+var files embed.FS
+
+// NewGenerator returns the generator to scaffold a new Cosmos SDK app.
+func NewGenerator(opts *Options) (*genny.Generator, error) {
+	// Remove "files/" prefix
+	subfs, err := fs.Sub(files, "files")
+	if err != nil {
+		return nil, fmt.Errorf("generator sub: %w", err)
+	}
+	g := genny.New()
```

Comment:
This is great - much clearer than the previous code.  We could do this in the rest of the codebase as well right?

If so, let's open an issue!

Reply:
Yes, this is totally applicable to other generators. I created the issue https://github.com/ignite/cli/issues/3393